packages/nimbus/src/components/date-picker/date-picker.stories.tsx

```diff
@@ -0,0 +1,2509 @@
+
+      // Focus first segment and type a value
+      await userEvent.click(firstSegment);
+      await userEvent.keyboard("12");
+
+      // Segment should have the typed value
+      await expect(firstSegment).toHaveAttribute("aria-valuenow", "12");
+
+      // Clear segments for next tests by selecting all and deleting
+      for (const segment of dateSegments) {
+        await userEvent.click(segment);
+        await userEvent.keyboard("{Control>}a{/Control}");
+        await userEvent.keyboard("{Delete}");
+      }
+    });
+
+    await step("Calendar popover opens and closes correctly", async () => {
+      const calendarButton = canvas.getByRole("button", { name: /calendar/i });
+
+      // Initially, calendar should not be visible (check in entire document since popover is portaled)
+      let calendar = within(document.body).queryByRole("application");
+      await expect(calendar).not.toBeInTheDocument();
+
+      // Click calendar button to open popover
+      await userEvent.click(calendarButton);
+
+      // Wait for calendar to appear (it's rendered in a portal)
+      await waitFor(async () => {
+        calendar = within(document.body).queryByRole("application");
+        await expect(calendar).toBeInTheDocument();
+      });
+
+      // Should be able to close with Escape key
+      await userEvent.keyboard("{Escape}");
+
+      // Wait for calendar to disappear
+      await waitFor(async () => {
+        calendar = within(document.body).queryByRole("application");
+        await expect(calendar).not.toBeInTheDocument();
+      });
+    });
+
+    await step("Clear button functionality works correctly", async () => {
+      const dateSegments = canvas.getAllByRole("spinbutton");
+
+      // Initially, clear button should be disabled (no value selected)
+      const clearButton = canvas.getByRole("button", { name: /clear/i });
+      await expect(clearButton).toBeDisabled();
+
+      // Set a date value first by changing from placeholder values
+      await userEvent.click(dateSegments[0]); // month
+      await userEvent.keyboard("1");
+      if (dateSegments[1]) {
+        await userEvent.click(dateSegments[1]); // day
+        await userEvent.keyboard("15");
+      }
+      if (dateSegments[2]) {
+        await userEvent.click(dateSegments[2]); // year
+        await userEvent.keyboard("2025");
+      }
+
+      // Clear button should now be enabled (actual date was entered)
+      await expect(clearButton).not.toBeDisabled();
+
+      // Click clear button to remove values
+      await userEvent.click(clearButton);
+
+      // After clearing, segments return to placeholder state (showing current date)
+      // The segments will still have values (current date placeholders) but DatePicker considers it "empty"
+      // Clear button should be disabled again indicating no actual value is selected
+      await expect(clearButton).toBeDisabled();
+    });
+
+    await step("Date selection from calendar works", async () => {
```

Comment:
Should we implement some of these tests in the `Calendar` story as well? I get that `Calendar` is well-tested by this file, but I'm hesitant to introduce a convention where contributors need to know that components that are children of a more complex component are tested in the more complex component.

I could go either way on this, really looking for thoughts more than asking for changes.